File: Data Structures & Algorithms/three-integer-sum/submission-2.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        var result = new List<List<int>>();
        var resultSet = new HashSet<string>();

        int length = nums.Length;

        for (int i = 0; i < length; i++) {
            for (int j = i + 1; j < length; j++) {
                for (int k = j + 1; k < length; k++) {
                    if (nums[i] + nums[j] + nums[k] == 0) {
                        var triplet = new List<int>() { nums[i], nums[j], nums[k] };

                        triplet.Sort();

                        string key = string.Join(',', triplet);
                        if (!resultSet.Contains(key)) {
                            resultSet.Add(key);
                            result.Add(triplet);
                        }
                    }
                }
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-2.ts
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs: string[]): string[][] {
        const result = new Map<string, string[]>();

        strs.forEach((str) => {
            const key = [...str].sort().join("");
            result[key] ??= [];
            result[key].push(str);
        });

        return Object.values(result);
    }
}
